File: 4.PatternMatching/PatternMatching.py
def pattern_matching(pattern: str, genome: str):
    """Find all occurrences of a pattern in a genome."""
    ind = []
    for i in range(len(genome) - len(pattern) + 1):
        if pattern == genome[i:i+len(pattern)]:
            ind.append(i)
    return ind



pat = "ATAT"
genome = "GATATATGCATATACTT"


print(pattern_matching(pat, genome))

File: 4.PatternMatching/PatternMatching.go
package main

import "fmt"

func PatternMatching(pattern, genome string) []int {
	var array []int
	for i := 0; i < len(genome)-len(pattern)+1; i++ {
		if pattern == genome[i:i+len(pattern)] {
			array = append(array, i)
		}
	}
	return array
}

func prt_array(arr []int) {
	for i := 0; i < len(arr); i++ {
		fmt.Print(arr[i], " ")
	}
}

func main() {
	pat := "ATAT"
	genome := "GATATATGCATATACTT"
	result := PatternMatching(pat, genome)
	prt_array(result)
}
